demos/device-shadow-for-aws-iot-embedded-sdk/shadow_demo_helpers.c

```diff
@@ -0,0 +1,916 @@
+                           outgoingPublishPackets[ ucIndex ].packetId ) );
+            }
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+TransportSocketStatus_t EstablishMqttSession( MQTTEventCallback_t eventCallback )
+{
+    TransportSocketStatus_t returnStatus = EXIT_SUCCESS;
+    MQTTStatus_t eMqttStatus;
+    MQTTConnectInfo_t xConnectInfo;
+    TransportInterface_t xTransport;
+    MQTTContext_t * pxMqttContext = &mqttContext;
+    NetworkContext_t * pxNetworkContext = &networkContext;
+    bool sessionPresent = false;
+
+    assert( pxMqttContext != NULL );
+    assert( pxNetworkContext != NULL );
+
+    /* Initialize the mqtt context and network context. */
+    ( void ) memset( pxMqttContext, 0U, sizeof( MQTTContext_t ) );
+    ( void ) memset( pxMqttContext, 0U, sizeof( NetworkContext_t ) );
+
+    returnStatus = prvConnectToServerWithBackoffRetries( pxNetworkContext );
+
+    if( returnStatus != EXIT_SUCCESS )
+    {
+        /* Log error to indicate connection failure after all
+         * reconnect attempts are over. */
+        LogError( ( "Failed to connect to MQTT broker %.*s.",
+                    strlen( democonfigMQTT_BROKER_ENDPOINT ),
+                    democonfigMQTT_BROKER_ENDPOINT ) );
+    }
+    else
+    {
+        /* Fill in Transport Interface send and receive function pointers. */
+        xTransport.pNetworkContext = pxNetworkContext;
+        xTransport.send = SecureSocketsTransport_Send;
+        xTransport.recv = SecureSocketsTransport_Recv;
+
+        /* Initialize MQTT library. */
+        eMqttStatus = MQTT_Init( pxMqttContext,
+                                 &xTransport,
+                                 prvGetTimeMs,
+                                 eventCallback,
+                                 &xBuffer );
+
+        if( eMqttStatus != MQTTSuccess )
+        {
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "MQTT init failed with status %u.", eMqttStatus ) );
+        }
+        else
+        {
+            /* Establish MQTT session by sending a CONNECT packet. */
+
+            /* Many fields not used in this demo so start with everything at 0. */
+            ( void ) memset( ( void * ) &xConnectInfo, 0x00, sizeof( xConnectInfo ) );
+
+            /* Start with a clean session i.e. direct the MQTT broker to discard any
+             * previous session data. Also, establishing a connection with clean session
+             * will ensure that the broker does not store any data when this client
+             * gets disconnected. */
+            xConnectInfo.cleanSession = true;
+
+            /* The client identifier is used to uniquely identify this MQTT client to
+             * the MQTT broker. In a production device the identifier can be something
+             * unique, such as a device serial number. */
+            xConnectInfo.pClientIdentifier = democonfigCLIENT_IDENTIFIER;
```

Comment:
I believe this is the same as the thing name in AWS IoT.

Reply:
yes, the configuration will be the same as clientcredentialIOT_THING_NAME.